ecs_model_deployer/src/index.ts

```diff
@@ -0,0 +1,130 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { spawnSync, spawn } from 'child_process';
+import { readdirSync, symlinkSync, rmSync } from 'fs';
+
+const ACTION_DEPLOY = 'deploy';
+const ACTION_DESTROY = 'destroy';
+
+/*
+  cdk CLI always wants ./ to be writable in order to write cdk.context.json.
+  This should really be an environment variable or something, but this function
+  should give us a writable ./ without wasting tons of compute time to copy
+*/
+const createWritableEnv = () => {
+    rmSync('/tmp/cdk.out', {recursive: true, force: true});
+    const files = readdirSync('.');
+    for ( const f of files ) {
+        /*
+          We want this file to be empty or absent so that any info needed is
+          pulled from the account instead of the cache
+        */
+        if ( f === 'cdk.context.json' ) {
+            continue;
+        }
+        const source = `/tmp/${f}`;
+        const target = `${process.env['PWD']}/${f}`;
+
+        try {
+            symlinkSync(target, source, 'file');
+        } catch ( err ) {
+            if ( err instanceof Error && err.message.match(/EEXIST/) ) {
+                // Writable env already established from previous call.
+            }
```

Comment:
the outcome is the same whether we hit the condition or not- should we throw another error, or just log that we're skipping if there's an exception?

Reply:
good catch, somehow left off the `else { throw err; }`. The intention of this was to eat the EEXIST error but let anything else throw. Will update this.